python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -631,34 +640,71 @@ def run(self, *, exposures, id_generator=None, result=None):
 
         result.exposure = self.snap_combine.run(exposures).exposure
         self._recordMaskedPixelFractions(result.exposure)
+        self.log.info("Initial PhotoCalib: %s", result.exposure.getPhotoCalib())
+
+        result.background = None
+        summary_stat_catalog = None
+        # Some exposure components are set to initial placeholder objects
+        # while we try to bootstrap them.  If we fail before we fit for them,
+        # we want to reset those components to None so the placeholders don't
+        # masquerade as the real thing.
+        have_fit_psf = False
+        have_fit_astrometry = False
+        have_fit_photometry = False
+        try:
```

Comment:
I note that you also asked @parejkoj about this block. I also can't think of a better alternative. My go-to would normally be to break this down into smaller sub-blocks, one for the PSF, one for photometry, one for astrometry, but that would involve a fair amount of logic when calling to ensure the prior step succeeded. With that in mind, I think I can go along with this large try/except block here too.